src/basic/FStar.Errors.fs

```diff
@@ -344,350 +360,408 @@ type raw_error =
+    Warning_TacAdmit                                  , CWarning, 296;
+    Fatal_IncoherentPatterns                          , CFatal, 297;
+    Error_NoSMTButNeeded                              , CAlwaysError, 298;
+    Fatal_UnexpectedAntiquotation                     , CFatal, 299;
+    Fatal_SplicedUndef                                , CFatal, 300;
+    Fatal_SpliceUnembedFail                           , CFatal, 301;
+    Warning_ExtractionUnexpectedEffect                , CWarning, 302;
+    Error_DidNotFail                                  , CAlwaysError, 303;
+    Warning_UnappliedFail                             , CWarning, 304;
+    Warning_QuantifierWithoutPattern                  , CSilent, 305;
+    Error_EmptyFailErrs                               , CAlwaysError, 306;
+    Warning_logicqualifier                            , CWarning, 307;
+    Fatal_CyclicDependence                            , CFatal, 308;
+    Error_InductiveAnnotNotAType                      , CError, 309;
+    Fatal_FriendInterface                             , CFatal, 310;
+    Error_CannotRedefineConst                         , CError, 311;
+    Error_BadClassDecl                                , CError, 312;
+    Error_BadInductiveParam                           , CFatal, 313;
+    Error_FieldShadow                                 , CFatal, 314;
+    Error_UnexpectedDM4FType                          , CFatal, 315;
+    Fatal_EffectAbbreviationResultTypeMismatch        , CFatal, 316;
+    Error_AlreadyCachedAssertionFailure               , CFatal, 317;
+    Error_MustEraseMissing                            , CWarning, 318;
+    Warning_EffectfulArgumentToErasedFunction         , CWarning, 319;
+    Fatal_EmptySurfaceLet                             , CFatal, 320;
+    Warning_UnexpectedCheckedFile                     , CWarning, 321;
+    Fatal_ExtractionUnsupported                       , CFatal, 322;
+    Warning_SMTErrorReason                            , CWarning, 323;
+    Warning_CoercionNotFound                          , CWarning, 324;
+    Error_QuakeFailed                                 , CError, 325;
+    Error_IllSMTPat                                   , CError, 326;
+    Error_IllScopedTerm                               , CError, 327;
+    Warning_UnusedLetRec                              , CWarning, 328;
+    Fatal_PolymonadicBind_conflict                    , CError, 329;
+    Warning_BleedingEdge_Feature                      , CWarning, 330;
+    Warning_IgnoredBinding                            , CWarning, 331;
+    Warning_AbstractQualifier                         , CWarning, 332;
+    Warning_CouldNotReadHints                         , CWarning, 333;
+    Warning_WarnOnUse                                 , CSilent, 334
   ]
-  (* Protip: if we keep the semicolon at the end, we modify exactly one
-   * line for each error we add. This means we get a cleaner git history/blame *)
+let max_error_number = 334
```

Comment:
duh, yes, thanks : ) (for max_number)

Regarding the second one: As things evolve, I think we will have gaps in error numbers. So, e.g., one can write `--warn_error X..Y` and not all numbers in [X..Y] will be valid error numbers. Should we insist that every number is valid? My take on this was "no" ... but perhaps we should. 

From that perspective, `--warn_error X` is just sugar for `--warn_error X..X` and so I didn't complain about that either. 

But happy to change the policy here ... what do you think we should do?

Reply:
ok, i'm changing it to enforce that all errors in the range are valid. If a user sees us reporting errors N and N+2, they should write `--warn_error N N+2`, rather than `--warn_error N..N+2` rather than relying on some unreliable knowledge that error N+1 doesn't exist.

Reply:
I wouldn't mind if if we succeeded when an error within the range does not exist... Otherwise removing an error from the list could cause build breakages downstream? But if that's the case then `lookup_error_range` needs some patching since it will currently explode if there's a gap in the list (I think).

(Also are error ranges actually used in practice? AFAIK we don't try to clump related errors together.)

Reply:
ah, this was the third issue I wanted to discuss at the F* meeting, but forgot to get to it.

Reply:
so, no errors for using --warn_error with a nonexistent issue number?

Reply:
IMO yes, no errors. (Just checked, it's also what OCaml does.)

Reply:
fixed. no errors.